src/main/kotlin/lotto/LottoMachine.kt

```diff
@@ -0,0 +1,94 @@
+package lotto
+
+class LottoMachine() {
+    private fun calculateTickets(userAmount: Int): Int {
+        val numberOfTickets = userAmount / LOTTO_PRICE
+        return numberOfTickets
+    }
+
+    private fun generateNumbers(): List<Int> {
+        val randomNumbers = (MIN_VAL..MAX_VAL).shuffled().take(6).sorted()
+        return randomNumbers
+    }
+
+    fun createTickets(userAmount: Int): List<Lotto> {
+        val lottos = mutableListOf<Lotto>()
+        val amountOfTickets = calculateTickets(userAmount)
+        var count = 0
+        while (count < amountOfTickets) {
+            val lotto = Lotto(generateNumbers())
+            lottos.add(lotto)
+            count++
+        }
+        return lottos
+    }
+
+    private fun compareTicketToWinningNumbers(
+        lotto: Lotto,
+        winningNumbers: List<String>,
+    ): Int {
+        var countMatches = 0
+        winningNumbers.forEach { number ->
+            if (number.toInt() in lotto.getNumbers()) {
+                countMatches++
+            }
+        }
+        return countMatches
+    }
+
+    private fun compareTicketToBonusNumber(
+        lotto: Lotto,
+        bonusNumber: Int,
+    ): Boolean {
+        return bonusNumber in lotto.getNumbers()
+    }
+
+    private fun createMap(): MutableMap<Rank, Int> {
+        val prizeCounter = mutableMapOf<Rank, Int>()
+        for (rank in Rank.values()) {
```

Comment:
```suggestion
        for (rank in Rank.entries) {
```

We can try using this instead.
I would like you to find the difference between these two and tell me about it :)